libs/lexical_cast/include/hpx/lexical_cast/bad_lexical_cast.hpp

```diff
@@ -0,0 +1,81 @@
+// Copyright Kevlin Henney, 2000-2005.
+// Copyright Alexander Nasonov, 2006-2010.
+// Copyright Antony Polukhin, 2011-2019.
+// Copyright Agustin Berge, 2019.
+//
+// SPDX-License-Identifier: BSL-1.0
+// Distributed under the Boost Software License, Version 1.0. (See
+// accompanying file LICENSE_1_0.txt or copy at
+// http://www.boost.org/LICENSE_1_0.txt)
+//
+// what:  lexical_cast custom keyword cast
+// who:   contributed by Kevlin Henney,
+//        enhanced with contributions from Terje Slettebo,
+//        with additional fixes and suggestions from Gennaro Prota,
+//        Beman Dawes, Dave Abrahams, Daryle Walker, Peter Dimov,
+//        Alexander Nasonov, Antony Polukhin, Justin Viiret, Michael Hofmann,
+//        Cheng Yang, Matthew Bradbury, David W. Birdsall, Pavel Korzh and other Boosters
+// when:  November 2000, March 2003, June 2005, June 2006, March 2011 - 2014
+
+#ifndef HPX_LEXICAL_CAST_BAD_LEXICAL_CAST_HPP
+#define HPX_LEXICAL_CAST_BAD_LEXICAL_CAST_HPP
+
+#include <hpx/config.hpp>
+
+#include <boost/throw_exception.hpp>
```

Comment:
Could we use our exception handling, instead?

Reply:
No, I don't think we can. The `errors` module should be a foundational module, but it has quite a few dependencies, some of which in turn depend on this module.

We could always replace it with a naked `throw`.